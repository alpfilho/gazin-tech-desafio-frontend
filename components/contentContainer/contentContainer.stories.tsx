import { ComponentMeta, ComponentStory } from '@storybook/react';

import { ContentContainer } from './contentContainer';

export default {
	title: 'Layout/Content Container',
	component: ContentContainer,
	parameters: {
		layout: 'fullscreen'
	}
} as ComponentMeta<typeof ContentContainer>;

export const Default: ComponentStory<typeof ContentContainer> = () => (
	<ContentContainer>
		<h1>Example Content</h1>
	</ContentContainer>
);
