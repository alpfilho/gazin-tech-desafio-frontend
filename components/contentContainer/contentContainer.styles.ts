import styled from 'styled-components';

export const Container = styled.div`
	width: 100%;
	height: auto;

	display: flex;
	justify-content: center;
	align-items: center;

	padding-left: 6em;
	padding-right: 6em;

	@media screen and (max-width: 992px) {
		padding-left: 2em;
		padding-right: 2em;
	}

	@media screen and (max-width: 768px) {
		padding-left: 1em;
		padding-right: 1em;
	}
`;

export const Content = styled.div`
	width: 100%;
	max-width: 1920px;
`;
