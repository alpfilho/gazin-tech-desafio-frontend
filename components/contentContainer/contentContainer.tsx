import React from 'react';

import { Container, Content } from './contentContainer.styles';

/**
 * Componente que mantém o conteúdo centralizado na tela,
 * limitando a expansão e aplicando margem
 */
export const ContentContainer: React.FC<{ className?: string }> = ({ children, className }) => (
	<Container className={className}>
		<Content>{children}</Content>
	</Container>
);
