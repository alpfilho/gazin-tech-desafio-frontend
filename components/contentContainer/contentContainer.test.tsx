import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './contentContainer.stories';

const { Default } = composeStories(stories);

describe('ContentContainer', () => {
	it('render a content container with example content', () => {
		render(<Default />);

		const exampleText = screen.getByText('Example Content');

		expect(exampleText).toBeInTheDocument();
		expect(exampleText).toBeVisible();
	});

	it('render content container unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
