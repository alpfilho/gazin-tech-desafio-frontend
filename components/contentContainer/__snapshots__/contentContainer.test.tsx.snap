// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer render content container unchanged 1`] = `
<div>
  <div
    class="Container-sc-yq2cgn-0 leaVgY"
  >
    <div
      class="Content-sc-yq2cgn-1 etbrxE"
    >
      <h1>
        Example Content
      </h1>
    </div>
  </div>
</div>
`;
